wcivf/apps/elections/import_helpers.py

```diff
@@ -24,29 +25,60 @@ def __init__(self, ee_helper=None):
         self.ee_helper = ee_helper
         self.election_cache = {}
 
+    def ballot_order(self, ballot_dict):
+        charisma_map = {
+            "ref": {"default": 100},
+            "parl": {"default": 90},
+            "europarl": {"default": 80},
+            "mayor": {"default": 70, "local-authority": 65},
+            "nia": {"default": 60},
+            "gla": {"default": 60, "a": 55},
+            "naw": {"default": 60, "r": 55},
+            "senedd": {"default": 60, "r": 65, "c": 60},
+            "sp": {"default": 60, "r": 55},
+            "pcc": {"default": 70},
+            "local": {"default": 40},
+        }
+        modifier = 0
+        ballot_paper_id = ballot_dict["ballot_paper_id"]
+
+        # Look up the dict of possible weights for this election type
+        weights = charisma_map.get(ballot_paper_id.split(".")[0], 30)
+
+        organisation_type = ballot_paper_id.split(".")[0]
+        default_weight_for_election_type = weights.get("default")
+        base_charisma = weights.get(
+            organisation_type, default_weight_for_election_type
+        )
+
+        # Look up `r` and `a` subtypes
+        subtype = re.match(r"^[^.]+\.([ar])\.", ballot_paper_id)
+        if subtype:
+            base_charisma = weights.get(subtype.group(1), base_charisma)
+
+        # by-elections are slightly less charismatic than scheduled elections
+        if ".by." in ballot_paper_id:
+            modifier += 1
+
+        return base_charisma - modifier
+
     def update_or_create_from_ballot_dict(self, ballot_dict):
         created = False
         slug = ballot_dict["election"]["election_id"]
 
+        election_weight = self.ballot_order(ballot_dict)
         if slug not in self.election_cache:
             election_type = slug.split(".")[0]
 
-            election_weight = 10
-            uses_lists = ballot_dict["election"]["party_lists_in_use"]
-            if uses_lists:
-                election_weight = 20
-            if election_type == "mayor":
-                election_weight = 5
-
             election, created = Election.objects.update_or_create(
                 slug=slug,
-                election_type=slug.split(".")[0],
+                election_type=election_type,
                 defaults={
                     "election_date": ballot_dict["election"]["election_date"],
                     "name": ballot_dict["election"]["name"].strip(),
                     "current": ballot_dict["election"]["current"],
                     "election_weight": election_weight,
-                    "uses_lists": uses_lists,
+                    "uses_lists": ballot_dict["party_lists_in_use"],
```

Comment:
Think this is missing the `["election"]` part - needs to be `ballot_dict["election"]["party_lists_in_use"]`